server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -145,7 +160,14 @@ protected void eventLoop() {
   protected void processNextEvent() {
     try {
       ShuffleDataFlushEvent event = flushQueue.take();
-      threadPoolExecutor.execute(() -> processEvent(event));
+      Storage storage = storageManager.selectStorage(event);
```

Comment:
It looks really strange for selecting storage here, which will select again before doing flush.

And the selected storage will be changed if writing failed. I think we'd better refactor flushing part.

Reply:
> It looks really strange for selecting storage here, which will select again before doing flush.
> 
> And the selected storage will be changed if writing failed. I think we'd better refactor flushing part.

You're right.

Reply:
I think that the storageType should not be switched in the flushToFile method. When the writing fails or the storage is )
corrupted, the event should be put into pendingEvents, and flushToFile should be performed again.

Reply:
+1.